beacon-chain/types/block.go

```diff
@@ -142,6 +143,24 @@ func (b *Block) Timestamp() (time.Time, error) {
 	return ptypes.Timestamp(b.data.Timestamp)
 }
 
+// Score returns the block's determined score based on its associated
+// crystallized state's last_finalized and last_justified slots used in the fork choice
+// rule of the beacon chain.
+func (b *Block) Score() uint64 {
+	return b.data.GetScore()
+}
+
+// SetScore determines the weighted block score utilized by the fork choice rule of the beacon chain.
+// This score currently depends on the last finalized and last justified slots of the
+// block's associated crystallized state as well as the block's slot number.
+// Currently, this uses the block's slot number as a factor in the calculations but the final, production
+// fork choice scoring rule may utilize the number of votes in the block.
+func (b *Block) SetScore(lastFinalizedSlot uint64, lastJustifiedSlot uint64) uint64 {
+	score := lastFinalizedSlot*uint64(math.Pow(10, 20)) + lastJustifiedSlot*uint64(math.Pow(10, 10)) + b.SlotNumber()
```

Comment:
This looks like an integer overflow. Max value for `uint64` should be 18446744073709551615, which is less than `math.Pow(10,20)`. The other issue is that we don't want to store the score in the block, since that would cause the blockhash to be a different value.